src/web/lib/contributions.ts

```diff
@@ -149,3 +150,20 @@ export const getArticleCountConsent = (): Promise<boolean> => {
         });
     });
 };
+
+export const withinLocalNoBannerCachePeriod = (): boolean => {
+    const item = window.localStorage.getItem(NO_BANNER_UNTIL_LATER_KEY);
+    if (item && !Number.isNaN(parseInt(item, 10))) {
+        const noBanner = parseInt(item, 10) > Date.now();
+        if (!noBanner) {
+            // Expired
+            window.localStorage.removeItem(NO_BANNER_UNTIL_LATER_KEY);
+        }
+        return noBanner;
+    }
+    return false;
+};
+
+const twentyMins = 20*60000;
+export const setLocalNoBannerCachePeriod = (): void =>
+    window.localStorage.setItem(NO_BANNER_UNTIL_LATER_KEY, `${Date.now() + twentyMins}`);
```

Comment:
minor point given 20 minutes, but it might be better to make this the last banner refreshed time, and then add the 20 minutes when checking (on line 157).  This just means if you want to change the period it takes effect as soon as you deploy, rather than after the caching period.

Reply:
I like this

Reply:
done!